subworkflows/local/virusbreakend_calling/main.nf

```diff
@@ -0,0 +1,154 @@
+//
+// VIRUSBreakend and Virus Interpreter identify viral content and insertion sites
+//
+
+import Constants
+import Utils
+
+include { VIRUSBREAKEND    } from '../../../modules/local/virusbreakend/main'
+include { VIRUSINTERPRETER } from '../../../modules/local/virusinterpreter/main'
+
+workflow VIRUSBREAKEND_CALLING {
+    take:
+    // Sample data
+    ch_inputs              // channel: [mandatory] [ meta ]
+    ch_tumor_bam           // channel: [mandatory] [ meta, bam, bai ]
+    ch_purple              // channel: [mandatory] [ meta, purple_dir ]
+    ch_bamtools_somatic    // channel: [mandatory] [ meta, metrics ]
+
+    // Reference data
+    genome_fasta           // channel: [mandatory] /path/to/genome_fasta
+    genome_fai             // channel: [mandatory] /path/to/genome_fai
+    genome_dict            // channel: [mandatory] /path/to/genome_dict
+    genome_gridss_index    // channel: [mandatory] /path/to/genome_gridss_index
+    virusbreakenddb        // channel: [mandatory] /path/to/virusbreakenddb/
+    virus_taxonomy_db      // channel: [mandatory] /path/to/virus_taxonomy_db
+    virus_reporting_db     // channel: [mandatory] /path/to/virus_reporting_db
+
+    // Params
+    gridss_config          // channel: [optional] /path/to/gridss_config
+
+    main:
+    // Channel for version.yml files
+    // channel: [ versions.yml ]
+    ch_versions = Channel.empty()
+
+    // Sort inputs
+    // NOTE(SW): VIRUSBreakend inputs are not allowed in the samplesheet, so aren't considered
```

Comment:
Is there a reason why not?

Reply:
The reason is that Virus Interpreter is always run immediately following VIRUSBreakend, so where a user has VIRUSBreakend output from oncoanalyser they will also have Virus Interpreter output. I decided to simplify this subworkflow by only allowing user-defined Virus Interpreter inputs since there would be no meaningful impact on users.